web/packages/teleterm/src/services/tshd/types.test.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { createInsecureClientCredentials } from 'teleterm/services/grpcCredentials';
+
+import createClient from './createClient';
+
+// This test detects situations where one of the generated JS protobufs has missing dependencies.
+// Dependencies must be provided as `--path` values to `buf generate` in build.assets/genproto.sh.
+test('generated protos import necessary dependencies', () => {
+  expect(() => {
+    createClient('localhost:0', createInsecureClientCredentials());
```

Comment:
Won't the tests fail anyway if a protobuf file is missing? 
Also, it seems to me that this `expect` won't be even run since the test suite fails on resolving modules.

I commented out some lines in `genproto.sh` and I got a lot of errors like that:
```
  ● Test suite failed to run

    Cannot find module '../../../../teleport/accesslist/v1/accesslist_pb.js' from 'gen/proto/js/teleport/lib/teleterm/v1/service_pb.js'

    Require stack:
      gen/proto/js/teleport/lib/teleterm/v1/service_pb.js
      web/packages/teleterm/src/services/tshd/types.ts
      web/packages/teleterm/src/services/tshd/testHelpers.ts
      web/packages/teleterm/src/services/tshd/fixtures/mocks.ts
      web/packages/teleterm/src/ui/fixtures/mocks.ts
      web/packages/teleterm/src/ui/AppInitializer/showStartupModalsAndNotifications.test.ts

      16 | var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);
      17 |
    > 18 | var teleport_accesslist_v1_accesslist_pb = require('../../../../teleport/accesslist/v1/accesslist_pb.js');
         |                                            ^
      19 | goog.object.extend(proto, teleport_accesslist_v1_accesslist_pb);
      20 | var teleport_lib_teleterm_v1_access_request_pb = require('../../../../teleport/lib/teleterm/v1/access_request_pb.js');
      21 | goog.object.extend(proto, teleport_lib_teleterm_v1_access_request_pb);

      at Resolver._throwModNotFoundError (node_modules/jest-resolve/build/resolver.js:427:11)
      at Object.require (gen/proto/js/teleport/lib/teleterm/v1/service_pb.js:18:44)
      at Object.require (web/packages/teleterm/src/services/tshd/types.ts:33:1)
      at Object.require (web/packages/teleterm/src/services/tshd/testHelpers.ts:19:1)
      at Object.require (web/packages/teleterm/src/services/tshd/fixtures/mocks.ts:19:1)
      at Object.require (web/packages/teleterm/src/ui/fixtures/mocks.ts:20:1)
      at Object.require (web/packages/teleterm/src/ui/AppInitializer/showStartupModalsAndNotifications.test.ts:19:1)
```

Reply:
This `expect` is bogus. Jest forces you to have at least one expect and we want to have a test that explicitly triggers a code path which imports some stuff from the protos.

That's also the reason I haven't checked if there are any existing tests which fail if some protos have incorrect deps. Maybe there are, but none of those tests is explicit about it.

Ideally we wouldn't even have to run tests to detect that missing files are an issue. However, the program we use for generating JS protos sticks `/* eslint-disable */` and `// @ts-nocheck` at the top of the file, so we can't really use either ESLint or TypeScript to detect that.

The biggest improvement in this PR is that UI tests are now triggered on changes in `gen/proto/js`. Matter of fact, now that I think about it, we should trigger UI lint (which does type checking) on changes to `gen/proto/js` as well, shouldn't we? This way if some code depends directly on types from protos and they get changed but nothing in `web` changes, we'll get type errors on the PR level as well.

Reply:
>Matter of fact, now that I think about it, we should trigger UI lint (which does type checking) on changes to gen/proto/js as well, shouldn't we? 

Makes sene to me.